libr/util/print.c

```diff
@@ -1263,7 +1269,7 @@ R_API void r_print_hexdump(RPrint *p, ut64 addr, const ut8 *buf, int len, int ba
 			bool eol = false;
 			if (!eol && p && p->flags & R_PRINT_FLAGS_REFS) {
 				ut64 off = 0;
-				if (i + 8 < len) {
+				if (i + 8 <= len) {
```

Comment:
this looks wrong. The for-loop does `i < len`, so I think there shouldn't be `<=` here.

Reply:
this is fixing the broken test

Reply:
It looks wrong anyway and I guess it will be detect by ASAN if you try the test with it.

Reply:
asan is not complaining here, but it was complaining in other places. ive added more tests and fixed the bugs i spotted by undefiuned behaviour and out of bounds accesses. current code is safer, but still needs more love. as longas i have two prs for pxr i would love to get them merged before going further to avoid more conflicts between them

Reply:
fixed